File: ios/A11yEventEmitter.m
#import "A11yEventEmitter.h"
#import <React/RCTBridgeModule.h>

@implementation A11yEventEmitter

RCT_EXPORT_MODULE();

- (NSArray<NSString *> *)supportedEvents {
  return @[@"onAccessibilityEvent"];
}

@end

File: ios/A11yEventEmitter.swift
import Foundation
import React

@objc(A11yEventEmitter)
class A11yEventEmitter: NSObject {

  // Exposing a method to JS
  @objc func sendAccessibilityEvent(_ message: String) {
    if let bridge = self.bridge {
      bridge.eventDispatcher().sendAppEvent(withName: "onAccessibilityEvent", body: ["message": message])
    }
  }

  // Exposing the module to JS
  @objc static func requiresMainQueueSetup() -> Bool {
    return true
  }

  @objc func constantsToExport() -> [AnyHashable: Any]! {
    return [:]
  }
}
